parser/src/main/java/com/linkedin/avroutil1/parser/AvscParseContext.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.parser;
+
+import com.linkedin.avroutil1.model.AvroNamedSchema;
+import com.linkedin.avroutil1.model.AvroSchema;
+import com.linkedin.avroutil1.parser.exceptions.AvroSyntaxException;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * maintains state for parse operations in progress - completed (parsed)
+ * schemas, unresolved references, current namespace etc
+ */
+public class AvscParseContext {
+    /**
+     * represents the resource being parsed (typically an avsc file or a raw avsc string)
+     */
+    protected final URI uri;
+    /**
+     * current (closest-defined) namespace. changes during parsing - pushed when a new
+     * namespace is defined, popped when leaving the scope of a defined namespace.
+     * initial value is [""]
+     */
+    protected final Deque<String> namespaceStack = new ArrayDeque<>();
+    /**
+     * all schemas defined (not simply referenced) within the current avsc being parsed
```

Comment:
yes. thats actually the common case for large schemas - avscs dont officially support imports, and so a single top level schemas may define 10s+ of names schemas (inner records etc). this would be all of them

Reply:
I always assumed avro supported imports. This will be good to call out in the field or class documentation.